### PR TITLE
Fetch PRs into local branches

### LINKED ISSUE #4: Identify relevant remotes/forks
This would be in the form of post-processing the archive file.  Given an archive, enumerate all unmerged PRs and find the repository and branch / commit for the most recent state of that PR.

### PR DESCRIPTION
**Pull request by MikeBishop**
Fixes #4.

### CODE REVIEW

File: pr-remotes.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    action="store_true",
+    help="Show what would be done, without making any changes"
+)
+parser.add_argument("repo", help="target GitHub repo")
+args=parser.parse_args()
+
+reference = archive_reference.loadReference(args.input, args.repo)
+
+target_prs = reference.prs.values() if args.merged else (pr for pr in reference.prs.values() if pr["state"] != "MERGED")
+commits = {pr["headRefOid"] + ":pr" + str(pr["number"]) for pr in target_prs}
```
Comment (martinthomson) [inline pr-remotes.py:27]: Other places I've seen this use "pulls/33".  This keeps the pull requests in a single directory under `.git/refs/remotes/origin/pulls/`.

**Review by martinthomson — COMMENTED** [review]
This is pretty useful.

Does the API provide any information about how to get information about the target of the pull request.  I've noticed that GitHub either tracks the shared common commit for the pull request and target branch, or the current state of the target branch (I'm not sure which) and only shows differences based on that commit.  New changes to the target branch aren't usually displayed.  It would be good to understand how we might consume information about pull requests, which includes where they were pointed at the time.

This is more important because GitHub saves comments relative to a diff between the pull request and its target branch, so knowing what that diff is would be good.  I think that this design is terrible, but it appears to be hard to rectify, so we might have to deal with the consequences of that choice.

**Comment by MikeBishop** [comment]
> Does the API provide any information about how to get information about the target of the pull request. I've noticed that GitHub either tracks the shared common commit for the pull request and target branch, or the current state of the target branch (I'm not sure which) and only shows differences based on that commit. New changes to the target branch aren't usually displayed. It would be good to understand how we might consume information about pull requests, which includes where they were pointed at the time.

I *think* that simply comes from git.  Given two commits, it can show you the things that are on the second history that aren't on the first, ignoring anything that is in the first but not the second.  It's essentially the same list that's used for a rebase.  Of course, that means for any given comment, we'll need to capture the commit it's pointing to.  And what about comments on commits that were rebased away?

File: pr-remotes.py (hunk 1 above)
Comment (MikeBishop) [reply pr-remotes.py:27]: Good suggestion.  Done.

**Pull request merged**